docassemble/assemblylinewizard/interview_generator.py

```diff
@@ -541,58 +540,60 @@ def source(self, follow_additional_fields=True, document_type="pdf"):
+                  # attachments is no longer always a DAList
+                  # TODO / FUTURE we could let this handle multiple forms at once
+                  for attachment in self.attachments:  # We will only have ONE attachment
+                      # TODO: if we really use multiple attachments, we need to change this
+                      # So there is a unique variable name
+                      content += "---\n"
+                      # Use a DADict to store the attachment here
+                      content += "objects:\n"
+                      # TODO: has_addendum should be a flag set in the generator, not hardcoded
+                      content += "  - " + self.attachment_variable_name + ': ALDocument.using(title="' + self.interview.description + '", filename="' + self.interview.file_name + '", enabled=True, has_addendum=False)\n'
+                      content += "---\n"
+                      content += "objects:\n"
+                      # TODO: 
+                      content += '  - al_user_bundle: ALDocumentBundle.using(elements=[' + self.attachment_variable_name + '], filename="' + self.interview.file_name + '.pdf", title="All forms to download for your records")' + '\n'
+                      content += '  - al_court_bundle: ALDocumentBundle.using(elements=[' + self.attachment_variable_name + '], filename="' + self.interview.file_name + '.pdf", title="All forms to download for your records")' + '\n'
+                      content += "---\n"
+                      content += "attachment:\n"
+                      content += "    variable name: " + self.attachment_variable_name + "[i]\n"
+                      content += "    name: " + oneline(attachment.name) + "\n"
+                      content += "    filename: " + varname(attachment.name).replace('_', '-') + "\n"
+                      if attachment.type == 'md':
+                          content += "    content: " + oneline(attachment.content) + "\n"
+                      elif attachment.type == 'pdf':
+                          content += "    skip undefined: True" + "\n"
+                          content += "    pdf template file: " + oneline(attachment.pdf_filename) + "\n"
+                          self.templates_used.add(attachment.pdf_filename)
+                          content += "    fields:" + "\n"
+                          for field in attachment.fields:
+                            content += field.attachment_yaml()
+                      elif attachment.type == 'docx':
+                          content += "    docx template file: " + oneline(attachment.docx_filename) + "\n"
+                          self.templates_used.add(attachment.docx_filename)
+                  done_with_content = True
             if not done_with_content:
-                content += "fields:\n"
+                content += "fields:\n"  # TODO: test removing \n here
```

Comment:
I do think we need this `\n`, but you're right, there is an extra one. It's being add in `field_entry_yaml()`, when `_max_length_str()` is empty. I'll make an issue for this an make a ~4 line fix.